lambda-runtime/src/AWS/Lambda/Client.hs

```diff
@@ -0,0 +1,192 @@
+
+  let statusCode = HTTP.responseStatus response
+
+  unless (HTTP.statusIsSuccessful statusCode) . throwError
+    $ UnExpectedRunTimeError statusCode "Could not retrieve next event"
+
+  return response
+  where
+    toNextEventRequest :: HTTP.Request -> HTTP.Request
+    toNextEventRequest req = req
+      { HTTP.path = "2018-06-01/runtime/invocation/next"
+      }
+
+sendEventSuccess
+  :: (MonadCatch m, MonadIO m)
+  => HTTPConfig
+  -> RequestId
+  -> JSON.Value
+  -> m ()
+sendEventSuccess HTTPConfig{..} requestId json = do
+  response <- catchConnectionError
+    . flip HTTP.httpNoBody manager
+    $ toEventSuccessRequest request
+
+  let statusCode = HTTP.responseStatus response
+
+  when (statusCode == HTTP.status413) . throwM . PayLoadTooLarge $ showc json
+
+  unless (HTTP.statusIsSuccessful statusCode) .
+    throwM $ UnExpectedRunTimeError statusCode "Could not post handler result"
+  where
+    toEventSuccessRequest :: HTTP.Request -> HTTP.Request
+    toEventSuccessRequest req = req
+      { HTTP.requestBody    = HTTP.RequestBodyLBS (JSON.encode json)
+      , HTTP.method         = "POST"
+      , HTTP.path           = "2018-06-01/runtime/invocation/" <> convert requestId <> "/response"
+      }
+
+sendEventError
+  :: forall e m . (Exception e, MonadIO m)
+  => HTTPConfig
+  -> RequestId
+  -> e
+  -> m ()
+sendEventError HTTPConfig{..} requestId error
+  = void
+  . liftIO
+  . flip HTTP.httpNoBody manager
+  $ toEventErrorRequest request
+  where
+    toEventErrorRequest :: HTTP.Request -> HTTP.Request
+    toEventErrorRequest req = (toBaseErrorRequest error req)
+      { HTTP.path = "2018-06-01/runtime/invocation/" <> convert requestId <> "/error"
+      }
+
+sendBootError :: MonadIO m => HTTPConfig -> InternalLambdaClientError -> m ()
+sendBootError HTTPConfig{..} error
+  = void
+  . liftIO
+  . flip HTTP.httpNoBody manager
+  $ toInitErrorRequest request
+  where
+    toInitErrorRequest :: HTTP.Request -> HTTP.Request
+    toInitErrorRequest req = (toBaseErrorRequest error req)
+      { HTTP.path = "2018-06-01/runtime/init/error"
+      }
+
+performHttpRequest :: HTTPConfig -> LambdaClient (HTTP.Response JSON.Value)
+performHttpRequest HTTPConfig{..} = do
+  response <- catchConnectionError $ HTTP.httpLbs request manager
+
+  body <- liftEither
+    . first (ResponseBodyDecodeFailure . convert)
+    . JSON.eitherDecode
+    $ HTTP.responseBody response
+
+  pure $ fmap (const body) response
+
+catchConnectionError :: (MonadCatch m, MonadIO m) => IO a -> m a
+catchConnectionError action =
+  catch (liftIO action)
+    $ \e -> throwM . ConnectionError $ (e :: HTTP.HttpException)
+
+toBaseErrorRequest :: Exception e => e -> HTTP.Request -> HTTP.Request
+toBaseErrorRequest (displayException -> error) req = req
+  { HTTP.requestBody    = HTTP.RequestBodyLBS (JSON.encode error)
+  , HTTP.method         = "POST"
+  , HTTP.requestHeaders =  [("Content-Type", "application/vnd.aws.lambda.error+json") ]
```

Comment:
__suggestion (nitpick):__ There is extra space after the `=` here, and inconsistent spacing before the `]` at the end.